libtenzir/src/node.cpp

```diff
@@ -284,6 +284,59 @@ auto node_state::get_endpoint_handler(const http_request_description& desc)
   return result->second;
 }
 
+auto spawn_components(node_actor::stateful_pointer<node_state> self) -> void {
+  using component_plugin_map = std::unordered_map<std::string, const component_plugin*>;
+  // 1. Collect all component_plugins into a name -> plugin* map:
+  component_plugin_map todo = {};
+  for (const auto* component : plugins::get<component_plugin>()) {
+    todo.emplace(component->component_name(), component);
+  }
+  // 2. Calculate an ordered loading sequnce based on the wanted_components of
+  //    each plugin.
+  std::vector<const component_plugin*> sequenced_components = {};
+  std::unordered_set<std::string> done = {};
+  auto derive_sequence = [&](auto derive_sequence, const std::string& name) {
+    auto entry = todo.find(name);
+    if (entry == todo.end()) {
+      TENZIR_WARN("{} is trying to load the unknown component {}, expect "
+                  "reduced functionality.",
+                  *self, name);
```

Comment:
I think this should either be a hard error or a debug message, but nothing in-between makes sense to me.